pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
Then don't we need to add it for all the other objects?  Watch?  Where does it end?  Seems like in the POST case you're creating something under a parent resource.

----- Original Message -----
> >  		ctx := scope.ContextFunc(req)
> >  		ctx = api.WithNamespace(ctx, namespace)
> > +		if len(name) > 0 {
> > +			ctx = api.WithValue(ctx, "name", name)
> 
> @smarterclayton - If that's the case :-) Could we just add a new interface
> that also takes the name? SubResourceCreater ? And then we just define a
> Create(ctx, name, obj) ?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7718/files#r29615049